cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -47,55 +49,58 @@ type Server struct {
 	globalPackagingNamespace string
 	globalPackagingCluster   string
 	kappClientsGetter        kappClientsGetter
-	defaultUpgradePolicy     upgradePolicy
+	pluginConfig             *kappControllerPluginParsedConfig
 }
 
 // parsePluginConfig parses the input plugin configuration json file and return the configuration options.
-func parsePluginConfig(pluginConfigPath string) (upgradePolicy, error) {
-	type kappControllerPluginConfig struct {
-		KappController struct {
-			Packages struct {
-				V1alpha1 struct {
-					DefaultUpgradePolicy string `json:"defaultUpgradePolicy"`
-				} `json:"v1alpha1"`
-			} `json:"packages"`
-		} `json:"kappController"`
-	}
-	var config kappControllerPluginConfig
+func parsePluginConfig(pluginConfigPath string) (*kappControllerPluginParsedConfig, error) {
+	// default configuration
+	config := defaultPluginConfig
 
-	pluginConfig, err := ioutil.ReadFile(pluginConfigPath)
+	// load the configuration file and unmarshall the values
+	pluginConfigFile, err := ioutil.ReadFile(pluginConfigPath)
+	if err != nil {
+		return config, fmt.Errorf("unable to open plugin config at %q: %w", pluginConfigPath, err)
+	}
+	var pluginConfig kappControllerPluginConfig
+	err = json.Unmarshal([]byte(pluginConfigFile), &pluginConfig)
 	if err != nil {
-		return none, fmt.Errorf("unable to open plugin config at %q: %w", pluginConfigPath, err)
+		return config, fmt.Errorf("unable to unmarshal pluginconfig: %q error: %w", string(pluginConfigFile), err)
 	}
-	err = json.Unmarshal([]byte(pluginConfig), &config)
+
+	// parse the configuration options
+	defaultUpgradePolicy := upgradePolicyMapping[pluginConfig.KappController.Packages.V1alpha1.DefaultUpgradePolicy]
+	defaultIncludePrereleases, err := strconv.ParseBool(pluginConfig.KappController.Packages.V1alpha1.DefaultUpgradePolicy)
```

Comment:
Is `pluginConfig.KappController.Packages.V1alpha1.DefaultUpgradePolicy` the right param to parse for `defaultIncludePrereleases`?

Reply:
Absolutely right! I didn't notice, thanks!